# Panera
MATLAB algorithm to reconstruct pan-genus metabolic model (PGMM) from the existing genome-scale metabolic models (GSMMs).
## Motivation
Genome-scale metabolic models (GSMMs) have emerged as a powerful tool for comprehending the complex relationship between an organism’s genetic makeup (genotype) and its observable traits (phenotype). These models are instrumental in simulating the physiological behaviour of the biological organisms in-silico and analyze their responses to varying nutrient conditions and genetic modifications. Prior research analyses have harnessed GSMMs to reconstruct microbial communities in silico, providing insights into their metabolic dependencies and functionalities. However, a major limitation of the GSMM-based community modelling approach exists while using 16S rRNA sequencing data due to their inherent ambiguity of lower taxonomic resolution. Given the widespread availability of 16S datasets, there is a pressing need for novel frameworks that can leverage this genus-level information along with species-level data. However, existing tools like “createPanModels” suffer from significant limitations. These limitations hinder the ability of the models to fully exploit the potential of 16S rRNA sequencing data. This tool relies on a simplistic merging of existing models, creating a lumped biomass and cannot leverage species-specific information in 16S data. Furthermore, these tools frequently restrict taxonomic input to either genus or species level. In this study, we propose a unique framework to address two critical challenges in microbial community characterization: (i) uncertainty associated with taxonomic assignment in 16S rRNA sequencing data and (ii) scarcity of representative genus-level models. We introduce _‘Panera’_, an innovative framework designed to model microbial communities under this uncertainty and yet perform metabolic inferences using pan-genus metabolic models (PGMMs). In essence,_‘Panera’_ emerges as a potent and effective approach to aid in metabolic modelling by enabling robust predictions of microbial community metabolic phenotypes, especially when dealing with limited 16S data, and offers insights into genus-level metabolic landscapes.


